pennylane/tape/qubit_param_shift.py

```diff
@@ -379,3 +379,22 @@ def processing_fn(results):
             return np.apply_along_axis(dot, 0, results)
 
         return tapes, processing_fn
+
+    @property
+    def specs(self):
+
+        if "grad_method" not in self._par_info[0]:
+            self._update_gradient_info()
+
+        # Initialize with the forward pass execution
+        num_executions = 1
+        # Loop over all variables
+        for _, info in self._par_info.items():
+
+            if info["grad_method"] == "A":
+                num_executions += len(info["op"].get_parameter_shift(info["p_idx"]))
+
+        info = super().specs
+        info["num_parameter_shift_executions"] = num_executions

```

Comment:
I am not sure I understand what is going on in this function. Maybe some more explicit comments could help?

Reply:
Any better?